packages/react-core/src/components/Alert/Alert.tsx

```diff
@@ -91,12 +100,32 @@ export const Alert: React.FunctionComponent<AlertProps> = ({
       setIsTooltipVisible(showTooltip);
     }
   }, [titleRef, truncateTitle, isTooltipVisible]);
-  const customClassName = css(
-    styles.alert,
-    isInline && styles.modifiers.inline,
-    variant !== AlertVariant.default && styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
-    className
-  );
+  React.useEffect(() => {
+    timeout = timeout === true ? 8000 : Number(timeout);
+    if (timeout > 0) {
+      const timer = setTimeout(() => setTimedOut(true), timeout);
+      return () => clearTimeout(timer);
+    }
+  }, []);
+  React.useEffect(() => {
+    const onDocumentFocus = () => {
+      if (divRef.current) {
+        if (divRef.current.contains(document.activeElement)) {
+          setContainsFocus(true);
+          setTimedOutAnimation(false);
+        } else if (containsFocus) {
+          setContainsFocus(false);
+          setTimeout(() => setTimedOutAnimation(true), timeoutAnimation);
+        }
+      }
+    };
+
+    document.addEventListener('focus', onDocumentFocus, true);
+
+    return () => document.removeEventListener('focus', onDocumentFocus, true);
+  }, [containsFocus]);
+  const dismissed = timedOut && timedOutAnimation && !isMouseOver && !containsFocus;
+  React.useEffect(onTimeout, [dismissed]);
```

Comment:
You should only call `onTimeout` when `dismissed` is `true` otherwise you notify on first render.
```suggestion
  React.useEffect(() => dismissed && onTimeout(), [dismissed]);
```